### PR TITLE
Adds workflow filter to advisor

### PR DESCRIPTION
**Pull request by AllenBW**
fixes https://issues.redhat.com/browse/ADVISOR-1257

yah. so now we'll have workloads on the urls, changes when global filter selection changes... not sure this work (that does the filtering) is in prod api yet... (results don't look different) :thinking:

also tired of excusing camel case, get over it eslint, we deal with it sometimes (makes this pr much larger than it woulda been by leaving it alone 🤷 )

### CODE REVIEW

File: src/PresentationalComponents/RulesTable/RulesTable.js [hunk 1]
```diff
@@ -73,13 +72,14 @@ const RulesTable = ({ rules, filters, rulesFetchStatus, setFilters, fetchRules,
     };
 
     const fetchRulesFn = useCallback(() => {
-        urlBuilder(filters, selectedTags);
-        const options = selectedTags.length && ({ tags: selectedTags.map(tag => encodeURIComponent(tag)).join('&tags=') });
+        urlBuilder(filters, selectedTags, workloads);
+        let options = selectedTags.length && ({ tags: selectedTags.map(tag => encodeURIComponent(tag)).join('&tags=') });
+        workloads !== undefined && (options = { ...options, ...workloads });
```
Comment (LI0131) [inline src/PresentationalComponents/RulesTable/RulesTable.js:77]: Could change this to `workloads && ...`

File: src/PresentationalComponents/SystemsTable/SystemsTable.js [hunk 2]
```diff
@@ -49,10 +48,10 @@ const SystemsTable = ({ systemsFetchStatus, fetchSystems, systems, intl, filters
     };
 
     const fetchSystemsFn = useCallback(() => {
-        const options = selectedTags.length && ({ tags: selectedTags && selectedTags.join() });
+        let options = selectedTags.length && ({ tags: selectedTags && selectedTags.join() });
+        workloads !== undefined && (options = { ...options, ...workloads });
```
Comment (LI0131) [inline src/PresentationalComponents/SystemsTable/SystemsTable.js:52]: Same as before `workloads && ...`

File: src/SmartComponents/Recs/Details.js [hunk 3]
```diff
@@ -51,7 +50,8 @@ const OverviewDetails = ({ match, fetchRuleAck, fetchTopics, fetchSystem, fetchR
 
     // eslint-disable-next-line react-hooks/exhaustive-deps
     const fetchRulefn = (newSort, rule = true, system = true) => {
-        const options = selectedTags !== null && selectedTags.length && ({ tags: selectedTags.map(tag => encodeURIComponent(tag)).join('&tags=') });
+        let options = selectedTags !== null && selectedTags.length && ({ tags: selectedTags.map(tag => encodeURIComponent(tag)).join('&tags=') });
+        workloads !== undefined && (options = { ...options, ...workloads });
```
Comment (LI0131) [inline src/SmartComponents/Recs/Details.js:54]: Here too 🤷‍♂️

File: src/SmartComponents/Topics/List.js [hunk 4]
```diff
@@ -9,20 +9,21 @@ import TopicsTable from '../../PresentationalComponents/TopicsTable/TopicsTable'
...
-import { isGlobalFilter } from '../../AppConstants';
 
 const TagsToolbar = asyncComponent(() => import(/* webpackChunkName: "TagsToolbar" */ '../../PresentationalComponents/TagsToolbar/TagsToolbar'));
 
-const List = ({ fetchTopics, intl, selectedTags }) => {
+const List = ({ fetchTopics, intl, selectedTags, workloads }) => {
     useEffect(() => {
-        const options = selectedTags !== null && selectedTags.length && ({ tags: selectedTags.join() });
+        let options = selectedTags !== null && selectedTags.length && ({ tags: selectedTags.join() });
+        workloads !== undefined && (options = { ...options, ...workloads });
```
Comment (LI0131) [inline src/SmartComponents/Topics/List.js:21]: Another `workloads && ...`

**Review by LI0131 — APPROVED** [review]
Awesome stuff @AllenBW 🏅
Gotta love 🐫 🧳

File: src/SmartComponents/Topics/List.js (hunk 4 above)
Comment (AllenBW) [reply src/SmartComponents/Topics/List.js:21]: good point.. less explicit, but cleaner... 👍

**Pull request merged**